sklearn/datasets/samples_generator.py

```diff
@@ -379,14 +379,15 @@ def sample_example():
     if not sparse:
         X = X.toarray()
 
-    if return_indicator:
-        lb = MultiLabelBinarizer()
+    # return_indicator can be True due to backward compatibility
+    if return_indicator in (True, 'sparse', 'dense'):
+        lb = MultiLabelBinarizer(sparse_output=(return_indicator == 'sparse'))
         Y = lb.fit([range(n_classes)]).transform(Y)
     else:
```

Comment:
It should raise an error if the user pass random string which says what are the possible value of return_indicator.